qemu/exec.c

```diff
@@ -955,6 +955,21 @@ void flatview_add_to_dispatch(struct uc_struct *uc, FlatView *fv, MemoryRegionSe
     register_subpage(uc, fv, &remain);
 }
 
+static ram_addr_t find_ram_offset_last(struct uc_struct *uc, ram_addr_t size)
+{
+    RAMBlock *block;
+    ram_addr_t result = 0;
+
+    RAMBLOCK_FOREACH(block) {
+        result = MAX(block->offset + block->max_length, result);
+    }
+
+    if (result + size > RAM_ADDR_MAX) {
+        abort();
```

Comment:
This is dangerous and should be avoided

Reply:
This is because `find_ram_offset()` has no sane way to indicate an error. Therefore `find_ram_offset_last()` behaves like `find_ram_offset()`. I currently don't see a better way to handle this case.